src/daft-local-execution/src/intermediate_ops/intermediate_op.rs

```diff
@@ -25,15 +25,23 @@ use crate::{
     },
 };
 
-pub enum IntermediateOperatorResult {
-    NeedMoreInput(Option<Arc<MicroPartition>>),
+/// Output states for intermediate operators
+pub enum IntermediateOperatorOutput {
+    /// Sink processed input and is yielding output,
+    /// ready for next input (stateless operation)
+    NeedMoreInput(Arc<MicroPartition>),
```

Comment:
Are we just removing the option because it's not being used? It may still be useful in the future, doesn't necessarily hurt to keep.